### ISSUE TITLE
Best Practice: Don't use the default storage account for heavy workloads

### ISSUE THREAD

**Issue by christopheranderson** [Azure/Azure-Functions#218]
Need to add a section in our docs on best practices, if we don't have one already. Specifically, we need to mention how one should not use the default storage account (AzureWebjobsStorage, AzureWebJobsDashboard) for your own function bindings if they'll be under any kind of heavy load.

Something like:

## Avoid heavy additional usage of the default Storage Account

It is a best practice to use a separate Storage Account for your Storage bindings. When using Storage Bindings (Blob, Queues, Tables), avoid using the default storage account (`AzureWebJobsStorage` & `AzureWebJobsDashboard`) when doing any high usage workloads (> 20 executions/second). This can lead to negative impact on the performance of the host, in addition to being throttled sooner by Storage due to the shared load of your bindings and the function host. If you're seeing high latency for your Storage bindings, errors in the portal about being unable to access your file system, or missing logs, you may want to check that you're not overloading your Azure Storage account.

**Comment by lindydonna** [comment]
@rachelappel Is this something you could take on?  Seems like a good fit for the best practices doc: https://docs.microsoft.com/en-us/azure/azure-functions/functions-best-practices

**Comment by rachelappel** [comment]
@lindydonna  Sure thing.

**Comment by MaximRouiller** [comment]
Is it still relevant? If yes, does that impact Durable Functions? Would Durable Functions have better performance if a separate storage account is created? 

cc @cgillum

**Comment by cgillum** [comment]
Yes, this is still relevant and yes, Durable Functions will generally perform better if a separate storage account is used. This is called out in our performance docs here: https://docs.microsoft.com/en-us/azure/azure-functions/durable-functions-perf-and-scale#storage-account-selection

**Comment by ColbyTresness** [comment]
@asavaritayal also a valid ask from a while ago

**Comment by fabiocav** [comment]
@asavaritayal @ggailey777 any comments here? Can we add a note on the best practices document?

**Comment by dcbrown16** [comment]
@asavaritayal is this still something we are looking to do? I have an advisory customer asking specifically for best practices regarding using the same Storage Account with multiple Functions.

**Comment by cgillum** [comment]
I think @ggailey777 is the right person to follow up on this.

**Comment by dcbrown16** [comment]
Somehow my posting here unassigned Asavari, so would definitely appreciate if @ggailey777 could weigh in.

**Comment by calloncampbell** [comment]
Any update on this? It would be nice to have this called out as a best practice.

**Comment by MaximRouiller** [comment]
I think this is covered here?

https://docs.microsoft.com/en-us/azure/azure-functions/functions-best-practices#avoid-sharing-storage-accounts

We could probably do more.

**Comment by tbresman** [comment]
I originally read the docs section (https://docs.microsoft.com/en-us/azure/azure-functions/functions-best-practices#avoid-sharing-storage-accounts) but felt a little unsure if I was understanding the recommendation correctly. I landed here after some more searching and I feel like the example in the original post was much more clear on the details.
